auth/src/secrets.rs

```diff
@@ -15,16 +15,27 @@ pub struct EdDsaManager {
 }
 
 impl EdDsaManager {
-    pub fn new() -> Self {
-        let doc = Ed25519KeyPair::generate_pkcs8(&ring::rand::SystemRandom::new())
-            .expect("to create a PKCS8 for edDSA");
-        let encoding_key = EncodingKey::from_ed_der(doc.as_ref());
-        let pair = Ed25519KeyPair::from_pkcs8(doc.as_ref()).expect("to create a key pair");
-        let public_key = pair.public_key();
+    // We take the private key used for JWT signing as an auth argument. It is the base64 enconding of a PEM encoded
+    // private key. The private key was generated previously using `OpenSSL 3.1.2 1 Aug 2023 (Library: OpenSSL 3.1.2 1
+    // Aug 2023)`, by using `openssl genpkey -algorithm ED25519`. This generates a PEM encoded PKCS#8 v1 formatted
+    // unencrypted key. We want to first parse the PEM encoded key and get the PKCS#8 v1 formatted key to parse it with
+    // `ring::signature::Ed25519KeyPair`. This is needed because the public key is not contained within the openssl
+    // generated private key, and should be computed/stored before serving it for JWT validation.
```

Comment:
What do you think of providing both the public and private key ? 

I'm asking because in [the documentation](https://docs.rs/ring/latest/ring/signature/struct.Ed25519KeyPair.html#method.from_pkcs8_maybe_unchecked), they're talking about a missing consistency check when using a PKCS#8 v1 file, since it doesn't have the public key, and while I don't know what impact it can have, the fact that they talk about it and that there exists a newer version of the file that fixes this would guide me towards preferring this solution. It is also exactly what the existing implementation is doing.